compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TaintAnalyzer.java

```diff
@@ -1517,6 +1595,9 @@ private void visitInvokable(BLangFunction invNode, SymbolEnv symbolEnv) {
                 // Set each parameter "tainted", then analyze the body to observe the outcome of the function.
                 analyzeReturnTaintedStatus(taintTable, invNode, symbolEnv, paramIndex, requiredParamCount,
                         defaultableParamCount);
+                if (analyzerPhase == AnalyzerPhase.LOOP_ANALYSIS_COMPLETE) {
+                    analyzerPhase = AnalyzerPhase.LOOP_ANALYSIS;
+                }
```

Comment:
/Can we avoid the duplicated code in this for loop?

Reply:
We cannot actually get rid of this check withing the for loop. This is essential for the loop analyzer to accurately work. 'analyzerPhase' will change to 'LOOP_ANALYSIS_COMPLETE' withing the 'analyzeReturnTaintedStatus' method, which is used to make sure regular analysis is done after the looping invocation is passed. However, we need the 'analyzeReturnTaintedStatus' to run similarly for each parameter to identify the tainted-status changes when each parameter is tainted. Hence, this check is essential.

Reply:
Agreed. Seems like the method was duplicated at some point to add an additional validation. Below method should be removed, will update:

`private void visitInvokable(BLangFunction invNode, SymbolEnv symbolEnv)`

Reply:
Just a note that this change was done.